foresight/utils/test_wrapper_utils.py

```diff
@@ -0,0 +1,266 @@
+
+    """
+        Foresight wrapper util class. It keeps all the generic information for execution of test. 
+        Creating test suite and test case, change application info, start and finish trace and invocation plugins.
+    """
+
+    __instance = None
+
+    @staticmethod
+    def get_instance(*args, **kwargs):
+        return TestWrapperUtils.__instance if TestWrapperUtils.__instance else TestWrapperUtils(*args, **kwargs) 
+
+
+    def __init__(self, *, plugin_executor=None, api_key=None, disable_trace=False, disable_metric=True, disable_log=True, opts=None, **ignored):
+        super(TestWrapperUtils, self).__init__(api_key, disable_trace, disable_metric, disable_log, opts)
+        ExecutionContextManager.set_provider(TracingExecutionContextProvider()) #TODO
+        self.application_info_provider = GlobalApplicationInfoProvider()
+        self._set_application_info("Foresight", "TestSuite", "TestSuite")
+        self.plugin_context = PluginContext(application_info=self.application_info_provider.get_application_info(),
+                                            request_count=0,
+                                            executor=plugin_executor,
+                                            api_key=self.api_key
+        )
+        max_test_log_count = ConfigProvider.get(config_names.THUNDRA_TEST_LOG_COUNT_MAX)
+        self.config.log_config = LogConfig(sampler=MaxCountAwareSampler(max_test_log_count))
+        self.plugins = wrapper_utils.initialize_plugins(self.plugin_context, disable_trace, disable_metric, disable_log,
+                                                        config=self.config)
+        TestWrapperUtils.__instance = self
+
+
+    def _set_application_info(self, application_class_name, application_domain_name, application_name):
+        try:
+            import pytest
+            self.application_info_provider.update({
+                "applicationClassName": application_class_name,
+                "applicationDomainName": application_domain_name,
+                "applicationName": application_name,
+                "applicationVersion": pytest.__version__
+            })
+            self.application_info_provider.update({
+                "applicationId": self._get_default_application_id()
+            })
+        except Exception as err:
+            logger.error("Test wrapper utils application info set error: {}".format(err))
+            pass
+
+
+    def _get_default_application_id(self):
+        try:
+            application_info = self.application_info_provider.get_application_info()
+            application_id = "python:test:pytest:{}:{}".format(application_info.get("applicationClassName"), application_info.get("applicationName"))
+            return application_id.lower()
+        except Exception as err:
+            logger.error("get default application id error: {}".format(err))
+            pass
+        return "python:test:pytest:dummy_application_id:{}".format(utils.create_uuid4())
```

Comment:
why `dummy_application_id` here?

Reply:
I have just put it to sign that there is an error while creating application id. Now, I have removed it.

Reply:
removed.